### DISCUSSION TITLE
Change versioning scheme

### DISCUSSION THREAD

**Discussion by logrusx** [focustimerhq/FocusTimer#705] [Ideas]
In the past everything was pretty simple, Gnome was just being updated, nothing broke, no big deal. In recent years however the Gnome team have introduced more and more incompatible changes. In the same time many distributions do not release a new version of the packages as soon as Gnome releases a version that breaks extensions and applications, so multiple incompatible versions of Gnome are being used out there.

Gnome Pomodoro has long surpassed the stage of beta software. It has been stable for many years. It doesn't need to have 0.x.x version any more.

What I propose is to adopt a new versioning scheme where the major version follows Gnome's major version, so that the minor versions and patch levels can be decoupled from Gnome's major versions and fixes and features can be independently incorporated into older versions of Gnome Pomodoro, rather than having to workarounds the version to incorporate a fix into an older version when the version after introduces incompatible change and there's no slot left.

**Comment by kamilprusko** [comment]
If the extension were distributed through extensions website again, in my opinion the versioning wouldn't matter. You would get the latest version that's compatible with your GNOME Shell version. Releases must go through approval process, so take into account some delay.

I am planning to release the upcoming GTK4 app as 1.0, as there are many breaking changes compared to 0.x. The extension can't be bundled with the app when distributed through Flatpak (offically), so the separation is inevitable...

**Comment by kamilprusko** [comment]
Reached 1.0-alpha, finally :D
